lib/taxjar/api/request.rb

```diff
@@ -72,17 +72,16 @@ def symbolize_keys!(object)
           object
         end
 
-        def fail_or_return_response_body(code, body)
-          e = extract_error(code, body)
-          fail(e) if e
-          body[object_key.to_sym]
-        end
-
-        def extract_error(code, body)
-          return Taxjar::Error.for_json_parse_error(code) if body.nil?
-          klass = Taxjar::Error::ERRORS[code]
-          if !klass.nil?
-            klass.from_response(body)
+        def fail_or_return_response_body(response, body)
+          if body.nil?
+            fail(Taxjar::Error.for_json_parse_error(response.code))
+          elsif response.status.success?
+            body[object_key.to_sym]
+          elsif !(klass = Taxjar::Error::ERRORS[response.code]).nil?
+            fail(klass.from_response(body))
+          else
+            message = HTTP::Response::Status::REASONS[response.code] || "Unknown Error"
+            fail(HTTP::Error, message)
```

Comment:
@poops Based on our documentation around [error handling](https://github.com/taxjar/taxjar-ruby#error-handling), should we consider raising a `Taxjar::Error` instead with additional details around `HTTP::Error` to ensure minimal changes by a developer using the API client?

Reply:
I wasn't sure since the [API docs](https://developers.taxjar.com/api/reference/#errors) state a specific list of errors, and some errors may not be coming from the API server.

Though based on the error handling you linked, I agree raising `Taxjar::Error` would be a safer bet.